pkg/controller/catalog/storage/mem_storage_test.go

```diff
@@ -20,278 +20,242 @@ import (
 	"strings"
 	"testing"
 
-	model "github.com/kubernetes-incubator/service-catalog/model/service_controller"
+	"github.com/kubernetes-incubator/service-catalog/pkg/apis/servicecatalog"
+	kapi "k8s.io/kubernetes/pkg/api"
 )
 
 const (
-	brokerOneUUID = "126b8154-a24a-4e79-9185-3df2eb4d18a8"
-	brokerTwoUUID = "2b0c42ed-c43a-4724-b883-e5ba878a8bfd"
+	brokerName1 = "Test1"
 )
 
 func TestNoBrokers(t *testing.T) {
-	s := CreateMemStorage()
-	l, err := s.ListBrokers()
+	const bogusBrokerName = "NOT THERE"
+	bs := CreateMemStorage().Brokers()
+	l, err := bs.List()
 	if err != nil {
-		t.Fatalf("ListBrokers failed with: %#v", err)
+		t.Fatalf("List failed with: %s", err)
 	}
 	if len(l) != 0 {
 		t.Fatalf("Expected 0 brokers, got %d", len(l))
 	}
-	b, err := s.GetBroker("NOT THERE")
+	b, err := bs.Get(bogusBrokerName)
 	if err == nil {
-		t.Fatal("GetBroker did not fail")
+		t.Fatal("Get did not fail")
 	}
 	if b != nil {
 		t.Fatalf("Got back a broker: %#v", b)
 	}
 }
 
 func TestAddBroker(t *testing.T) {
-	s := CreateMemStorage()
-	b := &model.ServiceBroker{GUID: "Test"}
-	cat := model.Catalog{
-		Services: []*model.Service{},
-	}
-	err := s.AddBroker(b, &cat)
+	bs := CreateMemStorage().Brokers()
+	b := &servicecatalog.Broker{ObjectMeta: kapi.ObjectMeta{Name: brokerName1}}
+	_, err := bs.Create(b)
 	if err != nil {
-		t.Fatalf("AddBroker failed with: %#v", err)
+		t.Fatalf("Create failed with: %s", err)
 	}
-	l, err := s.ListBrokers()
+	l, err := bs.List()
 	if len(l) != 1 {
 		t.Fatalf("Expected 1 broker, got %d", len(l))
 	}
-	b2, err := s.GetBroker("Test")
+	b2, err := bs.Get(brokerName1)
 	if err != nil {
-		t.Fatalf("GetBroker failed: %#v", err)
+		t.Fatalf("Get failed: %s", err)
 	}
 	if b2 == nil {
 		t.Fatal("Did not get back a broker")
 	}
-	if strings.Compare(b2.Name, b.Name) != 0 {
-		t.Fatalf("Names don't match, expected: '%s', got '%s'", b.Name, b2.Name)
+	if b2 != b {
```

Comment:
> I wasn't saying to check one field, rather all fields - deep copy.

Or "deep compare," as it were. I would still be concerned that many of the test resources used throughout the test suite look similar enough to one another that even a deep comparison could yield false negatives.

I _do_ agree with you that if the in-memory storage is ever to be used for anything else, it needs to be much more robust than it currently is. On the flip side of that, if it's just for testing (as I believe it is; and these in-memory impls are tantamount to "fakes"), then I think we shouldn't get too hung up on this, because this particular test is then literally testing other test code-- which seems a step too far.